api/open_saves.proto

```diff
@@ -89,6 +89,76 @@ service OpenSaves {
   // The string is optional and the server returns an empty string if
   // omitted.
   rpc Ping(PingRequest) returns (PingResponse) {}
+
+  // CompareAndSwapProperty compares the property to old_value and updates the property to value 
+  // if the old_value and the current property are equal.
+  // The updated field in CASPropertyResponse is set to true if the swap is executed.
+  // For example, CompareAndSwapProperty(property, value = 42, old_value = 24) will set the
+  // property to 42 if the current value is 24.
+  // CompareAndSwapProperty also supports swapping with a value of another type, e.g. 
+  // CompareAndSwapProperty(property, value = "42", old_value = 24).
+  // Otherwise it will not update the property and return the current (unchanged) value
+  // and updated = false. The operation is executed atomically.
+  // Errors:
+  //   - NotFound: the requested record or property was not found.
+  rpc CompareAndSwapProperty(CASPropertyRequest) returns (CASPropertyResponse) {}
+
+  // CompareAndSwapGreaterInt compares the number of an integer property to value and
+  // updates the property if the new value is greater than the current value.
+  // The updated field in AtomicResponse is set to true if the swap is executed.
+  // For example, CompareAndSwapGreaterInt(property, value = 42) will replace property with 42
+  // and return {value = old value, updated = true} if 42 > property.
+  // Otherwise it will not update the property and return the current (unchanged) value
+  // and updated = false. The operation is executed atomically.
+  // Errors:
+  //   - NotFound: the requested record or property was not found.
+  //   - InvalidArgument: the requested property was not an integer.
+  rpc CompareAndSwapGreaterInt(AtomicIntRequest) returns (AtomicIntResponse) {}
```

Comment:
In all these functions, I also would remove the Int. Not much value added in putting `Int` in the name since there is no other number types in the properties.
Same for the parameters: AtomicRequest, AtomicResponse.

Clear function names without abbreviation is often easier to remember.